core/src/main/java/pt/webdetails/cpf/utils/AbstractCorsUtil.java

```diff
@@ -0,0 +1,72 @@
+/*!
+ * Copyright 2002 - 2017 Webdetails, a Pentaho company. All rights reserved.
+ *
+ * This software was developed by Webdetails and is provided under the terms
+ * of the Mozilla Public License, Version 2.0, or any later version. You may not use
+ * this file except in compliance with the license. If you need a copy of the license,
+ * please go to http://mozilla.org/MPL/2.0/. The Initial Developer is Webdetails.
+ *
+ * Software distributed under the Mozilla Public License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. Please refer to
+ * the license for the specific language governing your rights and limitations.
+ */
+
+package pt.webdetails.cpf.utils;
+
+import org.apache.commons.lang.StringUtils;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Is used to implement cross-domain calls for an embedded scenario
+ */
+public abstract class AbstractCorsUtil {
+
+  private static final String TRUE = "true";
+  private static final String ORIGIN = "ORIGIN";
+  private static final String ACCESS_CONTROL_ALLOW_ORIGIN = "Access-Control-Allow-Origin";
+  private static final String ACCESS_CONTROL_ALLOW_CREDENTIALS = "Access-Control-Allow-Credentials";
+  private static final String COMMA = ",";
+
+  /**
+   * @return "true" if cross-domain calls are allowed
+   */
+  protected abstract String getAllowCrossDomainResources();
+
+  /**
+   * @return comma separated list of allowed domains
+   */
+  protected abstract String getDomainWhitelist();
+
+
+  public void setCorsHeaders( HttpServletRequest request, HttpServletResponse response ) {
+    final String allowCrossDomainResources = getAllowCrossDomainResources();
+    if ( allowCrossDomainResources != null && allowCrossDomainResources.equals( TRUE ) ) {
+      final String origin = request.getHeader( ORIGIN );
+      if ( origin != null && isDomainAllowed( origin ) ) {
+        response.setHeader( ACCESS_CONTROL_ALLOW_ORIGIN, origin );
+        response.setHeader( ACCESS_CONTROL_ALLOW_CREDENTIALS, TRUE );
+      }
+    }
+  }
+
+
+  private boolean isDomainAllowed( final String untrustedDomain ) {
+    final String pluginSetting = getDomainWhitelist();
+
+    if ( StringUtils.isNotBlank( pluginSetting ) ) {
+
+      final String[] domainList = pluginSetting.split( COMMA );
+
+      for ( String domain : domainList ) {
+        if ( domain.equals( untrustedDomain ) ) {
```

Comment:
The documentation seems to indicate that the `scheme` and `host` sections of URIs are case-insensitive, the value should be normalised to lower case, but it's not mandatory, so what if we make the comparison case insensitive?